packages/react/src/components/AccountSettings/types.ts

```diff
@@ -33,3 +33,7 @@ export type AccountSettingsError<Props = {}> = React.ComponentType<
 
 /** Form specific types */
 export type FormValues = Record<string, string>;
+
+export type BlurredFields = Record<string, boolean>;
```

Comment:
Couple things here:
- Would it make sense to track more than just whether or not the field has been blurred? Might be helpful to be able to look at whether or not a `field` has been `touched`, `dirty`, etc
- From what i can tell from reading the usage, we never assign any `false` values to `BlurredFields`

Reply:
> Would it make sense to track more than just whether or not the field has been blurred? 

IMO no, because `hasBlurred` is really all we need to determine whether to run validation or not. This is true with larger scope validation libraries too, such as react-hook-forms: their [skipValidation.ts](https://github.com/react-hook-form/react-hook-form/blob/master/src/logic/skipValidation.ts).

> From what i can tell from reading the usage, we never assign any false values to BlurredFields
No, it really depends on `falsey` values then gets set to `true`. Do you have any suggestion to improve this? 

I can set all of them to `false` initially, but I'm not sure that's necessary when we really only care about whether or not it's `true`.

Reply:
Changed to an array in [78e73c2](https://github.com/aws-amplify/amplify-ui/pull/2900/commits/78e73c27325e04b784154a116a8a85f9a99edde3).